### PR TITLE
feat(apim): Introduce gateway resource

### PR DESCRIPTION
**Pull request by tomkerkhove**
Introduce gateway resource for Azure API Management for **upcoming new API version, this is not going in to main**.

Things to point out:
- We have an active issue open to move to common error response model, but this new resource already supports it (https://github.com/Azure/azure-rest-api-specs/issues/27012)
- Any further issues related to this migration will be handled in a separate PR to the `release-ApiManagement-2023-09-01-preview` directly and not as part of this PR.

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary>Click here to see the details of Step 1, Breaking Changes review</summary>

If you are in purview of Step 1 of the diagram, follow the [Breaking Changes review process].  
**IMPORTANT!** This applies even if you believe your PR was mislabeled, for any reason, including tool failure.
</details>

<details> 
<summary>Click here to see the details of Step 2, ARM review</summary>

See https://aka.ms/azsdk/pr-arm-review.
</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [x] Other, please clarify:
    - Update API version that has not been released yet and is being prepared (see base branch)

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[Breaking Changes review process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by rkmanda** [comment]
Pl fix the errors flagged here or add a suppresion with an explanation for why they cant be fixed.

https://github.com/Azure/azure-rest-api-specs/pull/27757/checks?check_run_id=21655284311

Suppresions - https://github.com/Azure/autorest/blob/main/docs/generate/suppress-warnings.md

**Comment by rkmanda** [comment]
Note to next reviewer: Rest of the changes look good

**Comment by tomkerkhove** [comment]
@rkmanda Thanks for reviewing! These are known issues for which we've had exception in the past (https://github.com/Azure/azure-rest-api-specs/pull/26547#issuecomment-1839170944) and are tracked respectively in:
- https://github.com/Azure/azure-rest-api-specs/issues/27012
- https://github.com/Azure/azure-rest-api-specs/issues/23702

Given this PR is targeting release-ApiManagement-2023-09-01-preview and not main, is it OK to not have them as blocker?

However, I have made an update for the common types for errors given these are new resources which should be OK.

**Comment by rkmanda** [comment]
> @rkmanda Thanks for reviewing! These are known issues for which we've had exception in the past ([#26547 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/26547#issuecomment-1839170944)) and are tracked respectively in:
> 
> * [Move to common Model for Error Response  #27012](https://github.com/Azure/azure-rest-api-specs/issues/27012)
> * [APIM : No booleans only Enums  #23702](https://github.com/Azure/azure-rest-api-specs/issues/23702)
> 
> Given this PR is targeting release-ApiManagement-2023-09-01-preview and not main, is it OK to not have them as blocker?
> 
> However, I have made an update for the common types for errors given these are new resources which should be OK.

Note to reviewer: I removed the ARMCHangesRequested label because the author mentioned offline that they wanted this response to be looked at.

**Comment by raosuhas** [comment]
Looks like there are a couple Lint errors still which were not discussed above , specifically about havving the name and capacity as required fields in the PATCH body. Are you planning to fix those ?
:

**Comment by tomkerkhove** [comment]
@raosuhas I have missed those, sorry. Fixing these now:

**Comment by tomkerkhove** [comment]
I have updated this PR and we should be good to proceed.

Things to point out:
- We have an active issue open to move to common error response model, but this new resource already supports it (https://github.com/Azure/azure-rest-api-specs/issues/27012)
- Any further issues related to this migration will be handled in a separate PR to the `release-ApiManagement-2023-09-01-preview` directly and not as part of this PR.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-09-01-preview/apigateway.json [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+      "type": "object",
+      "properties": {
+        "minApiVersion": {
+          "type": "string",
+          "description": "Limit control plane API calls to API Management gateway with version equal to or newer than this value.",
+          "example": "2019-01-01"
+        }
+      },
+      "description": "Control Plane Apis version constraint for the API Management gateway."
+    },
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-09-01-preview/apigateway.json]: dont need this.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-09-01-preview/apigateway.json [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+        },
+        "capacity": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Capacity of the SKU (number of deployed units of the SKU)"
+        }
+      },
+      "required": [
+        "name",
+        "capacity"
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-09-01-preview/apigateway.json:500]: >"capacity" [](http://example.com/codeflow?start=8&length=10)

Based on spec only `Sku.name` is required

[https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/resource-api-reference.md#put-resource](https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/resource-api-reference.md#put-resource)

Comment (tomkerkhove) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-09-01-preview/apigateway.json:500]: Good point, updated

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-09-01-preview/apigateway.json (hunk 1 above)
Comment (tomkerkhove) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-09-01-preview/apigateway.json]: This was removed yesterday, thanks for confirming

**Review by ms-zhenhua — APPROVED** [review]

**Pull request merged**